server/text_generation_server/layers/fp8.py

```diff
@@ -40,16 +54,53 @@ def fp8_quantize(weight, qdtype=torch.float8_e4m3fn):
 class Fp8Linear(torch.nn.Module):
     def __init__(
         self,
-        weight,
+        qweight,
+        scale,
+        scale_upper_bound,
         bias,
+        dtype,
     ) -> None:
         super().__init__()
-        self.dtype = weight.dtype
-        self.qweight, self.scale = fp8_quantize(weight)
+        self.dtype = dtype
+        self.qweight = qweight
+        self.scale = scale
+        self.scale_upper_bound = scale_upper_bound
 
         self.bias = bias if bias is not None else None
 
+    @classmethod
+    def from_unquant(cls, weight, bias, dtype):
```

Comment:
Do we need to pass through `scale_upper_bound` here?

Reply:
I don't know from where we would get it from though.